core/cp4d_authenticator.go

```diff
@@ -63,6 +64,9 @@ type CloudPakForDataAuthenticator struct {
 	tokenData *cp4dTokenData
 }
 
+var cp4dMuOne sync.Mutex
+var cp4dMuTwo sync.Mutex
```

Comment:
I think it would be better to use meaningful names for these mutexes -- to be clear on what they lock/control.

I recommend:
- `cp4dMuOne` -> `cp4dRequestTokenMutex`
- `cp4dMuTwo` -> `cp4dNeedsRefreshMutex`